.github/workflows/release-container-image.yml

```diff
@@ -0,0 +1,237 @@
+      digest_arm64: ${{ steps.image.outputs.digest_arm64 }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Download digests
+        uses: actions/download-artifact@v4
+        with:
+          path: ${{ runner.temp }}/digests
+          pattern: digests-*
+          merge-multiple: true
+
+      - name: Login to GHCR
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          driver-opts: image=${{ env.BUILDKIT_IMAGE }}
+
+      - name: Create manifest list and push
+        working-directory: ${{ runner.temp }}/digests
+        run: |
+          DIGESTS=$(printf '${{ needs.prepare.outputs.image }}@sha256:%s ' *)
+          docker buildx imagetools create -t ${{ needs.prepare.outputs.image }} ${DIGESTS}
+
+      - name: Inspect image
+        id: image
+        run: |
+          # Inspect the image
+          docker buildx imagetools inspect ${{ needs.prepare.outputs.image }}
+          docker buildx imagetools inspect ${{ needs.prepare.outputs.image }} --format "{{json .Manifest}}" > manifest
+
+          # Calculate and print the digests
+          digest_root=$(jq -r .digest manifest)
+          digest_amd64=$(jq -r .manifests[0].digest manifest)
+          digest_arm64=$(jq -r .manifests[1].digest manifest)
+
+          echo "The image digests are:"
+          echo "  Root: $digest_root"
+          echo "  linux/amd64: $digest_amd64"
+          echo "  linux/arm64: $digest_arm64"
+
+          # NOTE: Set the digests as an output because the `env` context is not
+          # available to the inputs of a reusable workflow call.
+          echo "digest_root=$digest_root" >> "$GITHUB_OUTPUT"
+          echo "digest_amd64=$digest_amd64" >> "$GITHUB_OUTPUT"
+          echo "digest_arm64=$digest_arm64" >> "$GITHUB_OUTPUT"
+
+  # This step calls the container workflow to generate provenance and push it to
+  # the container registry.
+  provenance:
+    needs:
+      - prepare
+      - merge
+    strategy:
+      matrix:
+        digest:
+          - root
+          - amd64
+          - arm64
+    permissions:
+      actions: read # for detecting the Github Actions environment.
+      id-token: write # for creating OIDC tokens for signing.
+      packages: write # for uploading attestations.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@v2.0.0
+    with:
+      digest: ${{ needs.merge.outputs[format('digest_{0}', matrix.digest)] }}
+      image: ${{ needs.prepare.outputs.image }}
+      registry-username: ${{ github.actor }}
+    secrets:
+      registry-password: ${{ secrets.GITHUB_TOKEN }}
+
+  # This step ensures that the image is reproducible
+  check-reproducibility:
+    needs:
+      - prepare
```

Comment:
Should `prepare` be a requirement of `merge`?

Reply:
It is, but we need to define it here as well, in order to get the value with `${{ needs.prepare... }}`.

Reply:
Then maybe `merge` should pass the outputs so we can avoid this?

Reply:
Sure, we can proxy them in the `merge` job, but this way I fear that we may have to change things in two places in the future :face_with_diagonal_mouth:. If you feel it's better though, I can go ahead with that.

Reply:
Use your best judgement, I'm fine with both. If we let it like that, a comment would go a long way.

Reply:
Take a look at d7c01e755acd2f0f10aea4bd3d5975c7e4b34b99. If it feels weird, I'll just proxy the arguments.

Reply:
Yeah, it feels a bit weird to me, but I'm okay with it if you prefer it this way.

Reply:
Check out 2b6736f97856293fd8e4a4bea647e3b0bba3778b